src/rust/engine/engine_pyo3/src/externs/interface.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Do we want to change this file organization at all for our FFI? I copied and pasted `engine/src`'s layout, but we can use this as an opportunity to reorganize things.

For one, I think that `interface.rs` was way too big before, largely due to how hard it was with Rust-CPython to use types declared in other files. That is solved by PyO3 thanks to using normal Rust structs with normal visibility rules. We definitely want to break up `interface.rs`, although I'm not sure how? For example, `externs/interface/nailgun.rs`?

Reply:
Even if we do want to change it, it probably makes sense to have that reorganization be a separate PR.

Reply:
Generally, I would agree, but I think we can make some improvements along the way given this:

> Because the migration is incremental, we can more safely improve our FFI implementation along the way, rather than merely preserving the status quo. For example, this PR makes the PyStubCAS API more ergnonomic.